src/main/java/io/github/moulberry/notenoughupdates/profileviewer/hotm/HotmTreeScreen.kt

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2024 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.profileviewer.hotm
+
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonElement
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates
+import io.github.moulberry.notenoughupdates.autosubscribe.NEUAutoSubscribe
+import io.github.moulberry.notenoughupdates.core.GlScissorStack
+import io.github.moulberry.notenoughupdates.core.util.StringUtils
+import io.github.moulberry.notenoughupdates.events.RepositoryReloadEvent
+import io.github.moulberry.notenoughupdates.util.ItemUtils
+import io.github.moulberry.notenoughupdates.util.Utils
+import io.github.moulberry.notenoughupdates.util.kotlin.KotlinTypeAdapterFactory
+import moe.nea.lisp.*
+import moe.nea.lisp.bind.AutoBinder
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.ScaledResolution
+import net.minecraft.init.Items
+import net.minecraft.item.ItemStack
+import net.minecraft.util.ResourceLocation
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.math.floor
+import kotlin.math.roundToInt
+
+class HotmTreeRenderer(val hotmLayout: HotmTreeLayout, val prelude: List<String>) {
+    val lec = LispExecutionContext()
+
+    init {
+        lec.setupStandardBindings()
+        AutoBinder().bindTo(ExtraLispMethods(), lec.rootStackFrame)
+        prelude.forEachIndexed { index, it ->
+            lec.executeProgram(lec.rootStackFrame, LispParser.parse("hotmlayout.json:prelude[$index]", it))
+        }
+    }
+
+    @NEUAutoSubscribe
+    companion object {
+
+        val gson = GsonBuilder()
+            .registerTypeAdapterFactory(KotlinTypeAdapterFactory)
+            .registerTypeAdapterFactory(LoreLineSerializer)
+            .registerTypeAdapterFactory(LispProgramSerializer)
+            .create()
+
+        var renderer: HotmTreeRenderer? = null
+
+        @SubscribeEvent
+        fun onRepoReload(event: RepositoryReloadEvent) {
+            renderer = runCatching {
+                val hotmLayoutFile = NotEnoughUpdates.INSTANCE.manager.repoLocation
+                    .resolve("constants/hotmlayout.json")
```

Comment:
Why isnt this done the normal way in Constants.java?

Reply:
The normal way in Constants.java is bad. It just loads things into a JsonObject (which we shouldn't use raw like that anyway). There is just not much of a point in doing that when i will just load it into my own data structure anyways.